torch/_prims/utils.py

```diff
@@ -899,19 +996,46 @@ def wrap_device(d: Union[str, torch.device]) -> torch.device:
     return d
 
 
+# See implementation in TensorImpl.cpp
+# Returns true when a tensor is "dense" -- filling all of its storage
+# and "non-overlapping" -- each element in the tensor has a unique index.
+# This is equivalent to verifying that tensors with more than zero elements
+# have their sorted strides for dimensions of length greater than one
+# "nested" properly.
+def is_non_overlapping_and_dense(a: TensorLikeType) -> bool:
+    relevant_pairs = []
+    for x, y in zip(a.shape, a.stride()):
+        if x == 0:
+            return True
+        if x == 1:
+            pass
+        relevant_pairs.append((x, y))
+
+    expected = 1
+    for x, y in sorted(relevant_pairs, key=lambda p: p[1]):
```

Comment:
From a tracing perspective, this sort is terrifying

Reply:
Luckily the final version of the PR didn't include this function, and the sort in the stride comparison function shouldn't define any validity conditions, although it is an example of how we may have to just run the meta functions for our ops to understand what the intermediate metadata values of certain tensors are

Reply:
So, you are team "no symbolic strides"? Our current default assumption is that strides are symbolic, because from a design perspective that is easier. To make them not symbolic we will have to work (because strides are computed from symbolic quantities aka shapes).